### PR TITLE
Latest goflow

### PR DESCRIPTION
**Pull request by rowanseymour**
* use a broadcast for broadcast_created, a msg for msg_created
* add a fake channel to channel assets for simulator use so all messages get "sent"

### CODE REVIEW

File: temba/flows/models.py [hunk 1]
```diff
@@ -1034,7 +1034,7 @@ def handle_ruleset(cls, ruleset, step, run, msg, started_flows, resume_parent_ru
             value = msg.attachments[0].split(':', 1)[1]
 
         step.save_rule_match(rule, value)
-        ruleset.save_run_value(run, rule, value, msg.text)
+        ruleset.save_run_value(run, rule, value, six.text_type(msg))
```
Comment (rowanseymour) [inline temba/flows/models.py]: `@step.value` actually includes text and attachments - adjusting old engine to match results from new engine here

File: temba/utils/goflow/client.py [hunk 2]
```diff
@@ -263,16 +271,20 @@ def _request(self, endpoint, payload):
...
 
     if asset_type:
-        url = url + asset_type + '/'
+        url += asset_type + '/'
     if asset_uuid:
-        url = url + asset_uuid + '/'
+        url += asset_uuid + '/'
+
+    if simulator:
+        url += '?simulator=1'
```
Comment (rowanseymour) [inline temba/utils/goflow/client.py]: we don't want goflow to know the difference between a simulator session and a real session - so this means goflow will call back to rapidpro for assets with this in the URL to tell rapidpro to add the simulator channel

File: temba/flows/models.py (hunk 1 above)
Comment (nicpottier) [reply temba/flows/models.py]: This feels very omeletty

File: temba/flows/models.py [hunk 3]
```diff
@@ -2917,68 +2918,79 @@ def apply_events(self, run_log, msg_in=None):
...
+            media=media,
+            quick_replies=quick_replies,
+            base_language=event['base_language']
+        )
 
-        for recipient in itertools.chain(urns, contacts):
-            high_priority = (recipient == self.contact and self.session.responded)
-            channel = msg.channel if msg and msg.contact == recipient else None
+        # send in task
+        on_transaction_commit(lambda: send_broadcast_task.delay(broadcast.id, with_expressions=False))
```
Comment (nicpottier) [inline temba/flows/models.py]: Any reason to punt this to another task? We are already in a task here in all non-simulator cases no?

File: temba/utils/goflow/tests.py [hunk 4]
```diff
@@ -81,7 +70,10 @@ def test_add_contact_changed(self):
                     'name': 'Bob',
                     'language': None,
                     'timezone': 'UTC',
-                    'urns': ['twitterid:123456785?channel=%s#bobby' % str(twitter.uuid), 'tel:+12345670987'],
+                    'urns': [
+                        'twitterid:123456785?channel=%s#bobby' % str(twitter.uuid),
```
Comment (nicpottier) [inline temba/utils/goflow/tests.py]: Huh, didn't realize the fragment came after the query, guess that's how URLs do it by kinda hate it. Oh well.

File: temba/flows/models.py (hunk 3 above)
Comment (rowanseymour) [reply temba/flows/models.py]: if someone starts a contact flow which includes a broadcast to the rest of Nigeria... we want to carry on with the flow without waiting for that broadcast to be sent, no?

File: temba/flows/models.py (hunk 1 above)
Comment (rowanseymour) [reply temba/flows/models.py]: possibly... and it's still not even quite right, right? shouldn't it be whatever the operand of the ruleset was?

File: temba/flows/models.py (hunk 3 above)
Comment (nicpottier) [reply temba/flows/models.py]: Not positive actually, but I guess for the sake of committing the run/actions it is better to create them and have them applied async.

File: temba/flows/models.py (hunk 1 above)
Comment (nicpottier) [reply temba/flows/models.py]: This isn't actually changing `@step.value` though is it? It looks to me like `@step.value` in the old engine is actually the text and attachments of the last message. Rather this affects `@flow.age.text` in the old engine. (which is what `@flow.age.operand` in the new engine?)

Comment (rowanseymour) [reply temba/flows/models.py]: in old world currently:
```
@step.text == msg.text
@step.value == msg.text + msg.attachments
@flow.age.text == run.results['age']['input']
```

in new world:

```
@run.input = msg.text + msg.attachments
@run.input.text = msg.text
@run.results.age.input == router operand
```

Comment (nicpottier) [reply temba/flows/models.py]: Seems we should probably rename `@run.results.age.input` in new world to be operand. That operand may end up being `@run.input` in cases where you are splitting on that but that isn't guaranteed. Old world will always just be dumb and use `@run.input` for operand regardless.

**Pull request merged**